File: backend/internal/domain/graph.go
package domain

// GraphResponse represents the full story graph for Vue Flow
type GraphResponse struct {
	Story StorySummary    `json:"story"`
	Nodes []GraphNode     `json:"nodes"`
	Edges []GraphEdge     `json:"edges"`
}

// StorySummary represents minimal story info for graph response
type StorySummary struct {
	ID     string `json:"id"`
	Title  string `json:"title"`
	Status string `json:"status"`
}

// GraphNode represents a node in Vue Flow format
type GraphNode struct {
	ID       string      `json:"id"`
	Type     string      `json:"type"`
	Position Position    `json:"position"`
	Data     NodeData    `json:"data"`
}

// Position represents node position in Vue Flow
type Position struct {
	X float64 `json:"x"`
	Y float64 `json:"y"`
}

// NodeData represents the data payload for a scene node
type NodeData struct {
	Content string  `json:"content"`
	IsStart bool    `json:"is_start"`
	IsEnd   bool    `json:"is_end"`
	ImageURL *string `json:"image_url,omitempty"`
}

// GraphEdge represents an edge in Vue Flow format
type GraphEdge struct {
	ID     string `json:"id"`
	Source string `json:"source"`
	Target string `json:"target"`
	Label  string `json:"label"`
}

File: frontend/src/api/graph.ts
export interface StorySummary {
  id: string
  title: string
  status: string
}

export interface GraphNode {
  id: string
  type: string
  position: {
    x: number
    y: number
  }
  data: {
    content: string
    is_start: boolean
    is_end: boolean
    image_url?: string
  }
}

export interface GraphEdge {
  id: string
  source: string
  target: string
  label: string
}

export interface GraphResponse {
  story: StorySummary
  nodes: GraphNode[]
  edges: GraphEdge[]
}

const API_BASE_URL = (import.meta as any).env?.VITE_API_URL || 'http://localhost:8080'

async function fetchWithAuth(url: string, options: RequestInit = {}): Promise<Response> {
  const response = await fetch(`${API_BASE_URL}${url}`, {
    ...options,
    headers: {
      'Content-Type': 'application/json',
      ...options.headers,
    },
    credentials: 'include',
  })

  if (!response.ok) {
    const error = await response.json().catch(() => ({ error: 'Unknown error' }))
    throw new Error(error.error || `HTTP ${response.status}`)
  }

  return response
}

export const graphApi = {
  async loadGraph(storyId: string): Promise<GraphResponse> {
    const response = await fetchWithAuth(`/api/creator/stories/${storyId}/graph`, {
      method: 'GET',
    })
    return response.json()
  },
}
